cmd/common/oci.go

```diff
@@ -188,6 +188,24 @@ func NewRunCommand(rootCmd *cobra.Command, runtime runtime.Runtime, hiddenColumn
 				return fmt.Errorf("gadget instance id or name are ambiguous")
 			}
 			imageName = instances[0].Id
+
+			if len(instances[0].Nodes) > 0 {
+				// GetGadgetInfo can only be run on nodes that know about a gadget instance, so we need
+				// to make sure that the given nodes are valid
+				nodes := runtimeParams.Get(grpcruntime.ParamNode).AsStringSlice()
+				if len(nodes) == 0 {
+					nodes := strings.Join(nodes, ",")
```

Comment:
`nodes := strings.Join(instances[0].Nodes, ",")` Otherwise I get:

```bash
$ kubectl-gadget attach e9
Error: fetching gadget information: getting gadget info: rpc error: code = Unknown desc = instance e9e8dcb91d9c8b693396398478d11cf7 not found
```